src/Tizen.Wearable.CircularUI.Forms.Renderer/Shell/NavigationDrawer.cs

```diff
@@ -0,0 +1,312 @@
+
+            _gestureBox.RepeatEvents = true;
+
+            _outterBox = new Box(this);
+            _outterBox.Show();
+
+            _contentBox = new Box(this);
+            _contentBox.PropagateEvents = false;
+
+            _label = new ElmSharp.Label(_outterBox)
+            {
+                //TODO remove it
+                BackgroundColor = EColor.Pink,
+            };
+
+            _contentBox.Show();
+            _label.Show();
+
+            _outterBox.PackEnd(_contentBox);
+            _outterBox.PackEnd(_label);
+
+            _label.StackAbove(_contentBox);
+
+            _gestureLayer = new GestureLayer(_gestureBox);
+            _gestureLayer.Attach(_gestureBox);
+
+            _gestureLayer.SetMomentumCallback(GestureLayer.GestureState.Start, OnContentDragStarted);
+            _gestureLayer.SetMomentumCallback(GestureLayer.GestureState.End, OnContentDragEnded);
+            _gestureLayer.SetMomentumCallback(GestureLayer.GestureState.Abort, OnContentDragEnded);
+
+            _dragGestureLayer = new GestureLayer(_outterBox);
+            _dragGestureLayer.Attach(_outterBox);
+
+            _dragGestureLayer.SetMomentumCallback(GestureLayer.GestureState.Move, OnDrawerDragged);
+            _dragGestureLayer.SetMomentumCallback(GestureLayer.GestureState.End, OnDrawerDragEnded);
+            _dragGestureLayer.SetMomentumCallback(GestureLayer.GestureState.Abort, OnDrawerDragEnded);
+
+            _gestureBox.SetLayoutCallback(OnLayout);
+            _outterBox.SetLayoutCallback(OnContentLayout);
+
+            RotaryEventManager.Rotated += OnRotateEventReceived;
+        }
+
+        async void OnRotateEventReceived(EventArgs args)
+        {
+            if (_fadeInCancelTokenSource != null)
+            {
+                _fadeInCancelTokenSource.Cancel();
+            }
+
+            _fadeInCancelTokenSource = new CancellationTokenSource();
+            var token = _fadeInCancelTokenSource.Token;
+
+            if (!_isOpen)
+            {
+                await HideAsync(_outterBox);
+
+                _ = ShowAsync(_outterBox, cancelltaionToken:token);
+            }
+        }
+
+        void OnLayout()
+        {
+            var geometry = _parent.Geometry;
+            geometry.Y = (_isOpen) ? 0 : (geometry.Height - MinimumDrawerHeight);
+
+            _gestureBox.Geometry = _parent.Geometry;
+            _outterBox.Geometry = geometry;
+        }
+
+        void OnContentLayout()
+        {
+            var geometry = _outterBox.Geometry;
+            geometry.Height = MinimumDrawerHeight;
+
+            _contentBox.Geometry = _outterBox.Geometry;
+            _label.Geometry = geometry;
+        }
+
+        void OnContentDragStarted(GestureLayer.MomentumData moment)
+        {
+            if (_fadeInCancelTokenSource != null)
+            {
+                _fadeInCancelTokenSource.Cancel();
+                _fadeInCancelTokenSource = null;
+            }
+
+            _ = HideAsync((EvasObject)_outterBox);
```

Comment:
When drawer was opend, should not hide outterbox

Reply:
Fixed.